wolfssl/wolfcrypt/asn_public.h

```diff
@@ -187,8 +187,12 @@ enum {
 
     PEM_PASS_READ  = 0,
     PEM_PASS_WRITE = 1,
-};
+#if defined(NO_ASN)
```

Comment:
I don't see a reason to add an ifdef check around these. This is the only place they are defined. Let's say NO_ASN is defined and I still want to know `PKCS5_SALT_SZ` or `HMAC_SHA256_OID` for my own purposes? Also having them in the enum list does not increase code size.

Reply:
@dgarske, 

 error: redeclaration of enumerator ‘PKCS5_SALT_SZ’ and HMAC_SHA256_OID when building without

Reply:
@tmael : I don't see a redeclaration error with that configure. Also those two `PKCS5_SALT_SZ` and `HMAC_SHA256_OID` are only defined in asn_public.h.

Reply:
@dgarske, sorry try again with --disable-asn

Reply:
@tmael : `--disable-asn` is in the statement above. Can you confirm ./configure and post the errors?

Reply:
@dgarske 
```
  CC       wolfcrypt/src/src_libwolfssl_la-wolfmath.lo
wolfcrypt/src/wc_encrypt.c:243:25: error: unknown type name ‘EncryptedInfo’
 int wc_BufferKeyDecrypt(EncryptedInfo* info, byte* der, word32 derSz,
                         ^~~~~~~~~~~~~
wolfcrypt/src/wc_encrypt.c:306:25: error: unknown type name ‘EncryptedInfo’
 int wc_BufferKeyEncrypt(EncryptedInfo* info, byte* der, word32 derSz,
                         ^~~~~~~~~~~~~
wolfcrypt/src/wc_encrypt.c: In function ‘wc_CryptKey’:
wolfcrypt/src/wc_encrypt.c:429:22: error: ‘HMAC_SHA256_OID’ undeclared (first use in this function); did you mean ‘WC_SHA256’?
                 case HMAC_SHA256_OID:
                      ^~~~~~~~~~~~~~~
                      WC_SHA256
wolfcrypt/src/wc_encrypt.c:429:22: note: each undeclared identifier is reported only once for each function it appears in
Makefile:4494: recipe for target 'wolfcrypt/src/src_libwolfssl_la-wc_encrypt.lo' failed
make[1]: *** [wolfcrypt/src/src_libwolfssl_la-wc_encrypt.lo] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/home/tesfa/wolfssl'
Makefile:3031: recipe for target 'all' failed
make: *** [all] Error 2

```

Reply:
@tmael : The issue is `HMAC_SHA256_OID` and `PKCS5_SALT_SZ` are defined in both `asn_public.h` and `asn.h`.

The `PKCS5_SALT_SZ` is only used by wc_encrypt.c, so please move it to wc_encrypt.h.

The `HMAC_SHA256_OID` should only exist in asn.h, but that entire `enum HMAC_Sum` should be moved out of `NO_ASN`.

David